aziotctl/aziotctl-common/src/system/status.rs

```diff
@@ -8,140 +8,152 @@ use anyhow::{Context, Result};
+            service
+                .name
+                .strip_suffix(".service")
+                .unwrap_or(service.name)
+        );
+
+        print_logs(service.name, &service.state)?;
+        for socket in service.sockets {
+            if !matches!(socket.state, State::Active) {
+                print_logs(socket.name, &socket.state)?;
+            }
+        }
     }
 
-    Ok(())
-}
-
-fn read_status(process: &str) -> Result<Status> {
-    let result = Command::new("systemctl")
-        .args(&["is-active", process])
-        .output()
-        .context("Failed to call systemctl is-active")?;
-
-    let output = String::from_utf8_lossy(&result.stdout);
-
-    let result = match output.trim() {
-        "active" => Status::Active,
-        "inactive" => Status::Inactive,
-        _ => Status::Failed(format!(
-            "{} {}",
-            String::from_utf8_lossy(&result.stdout),
-            String::from_utf8_lossy(&result.stderr)
-        )),
-    };
-
-    Ok(result)
-}
-
-fn print_logs(process: &str, state: &Status) -> Result<()> {
+    let name = crate::program_name();
+    println!("Use '{} system logs' to check for non-fatal errors.", name);
     println!(
-        "{} is in a bad state: {:?}. Printing the last 10 log lines.",
-        process, state,
+        "Use '{} check' to diagnose connectivity and configuration issues.",
+        name
     );
-    Command::new("journalctl")
-        .args(&["-u", process, "--no-pager", "-e", "-n", "10"])
-        .spawn()
-        .context("Failed to spawn new process for printing logs")?
-        .wait()
-        .context("Failed to call journalctl")?;
-    println!();
 
     Ok(())
 }
 
-#[derive(PartialEq, Debug)]
-enum Status {
+#[derive(Clone, Debug)]
+enum State {
     Active,
-    Failed(String),
     Inactive,
+    Failed(String),
+}
+
+impl State {
+    fn from_systemctl(process: &str) -> Result<State> {
+        let result = Command::new("systemctl")
+            .args(&["is-active", process])
+            .output()
+            .context("Failed to call systemctl is-active")?;
+
+        let output = String::from_utf8_lossy(&result.stdout);
+
+        let result = match output.trim() {
+            "active" => State::Active,
+            "inactive" => State::Inactive,
+            _ => State::Failed(format!(
+                "{} {}",
+                String::from_utf8_lossy(&result.stdout).trim(),
+                String::from_utf8_lossy(&result.stderr).trim()
+            )),
+        };
+
+        Ok(result)
+    }
 }
 
-impl fmt::Display for Status {
+impl fmt::Display for State {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
-            Status::Active => write!(f, "active"),
-            Status::Inactive => write!(f, "inactive"),
-            Status::Failed(message) => write!(f, "{}", message),
+            State::Active => write!(f, "Running"),
+            State::Inactive => write!(f, "Ready"),
```

Comment:
This is different for services vs sockets. An inactive service is "Ready". An inactive socket is "Down".

Reply:
I feel like I may have gone a bit too fancy with the implementation. let me know if I should tone it down a bit

Reply:
Well, returning `Cow<'static, str>` would've been enough, but since you've already written this you may as well keep it.